api/v1alpha1/httproute_types.go

```diff
@@ -190,6 +200,75 @@ type HTTPRouteAction struct {
 	Extension *core.TypedLocalObjectReference `json:"extension"`
 }
 
+// TLSConfig defines the schema of a TLS configuration.
+type TLSConfig struct {
+	// Termination defines how to terminate TLS connections.
+	//
+	// If unspecified, TLS termination type "Edge" will be used.
+	//
+	// Support: core
+	//
+	// +optional
+	Termination TLSTerminationType `json:"termination,omitempty"`
+
+	// Certificate is a reference to a Kubernetes object containing
+	// the identity certificate, key and CA certificate used to terminate
```

Comment:
I think this needs to be specced in detail for implementations to be interoperable. I raised most of the issues that occurred to me in #74 . We should also consider whether to allow separation of the key from all the certificate material.